modules/azure/servicebus.go

```diff
@@ -0,0 +1,377 @@
+	nsClient, err := serviceBusNamespaceClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	iteratorNamespaceRules, err := nsClient.ListAuthorizationRulesComplete(
+		context.Background(), resourceGroup, namespace)
+
+	if err != nil {
+		return nil, err
+	}
+
+	results := []string{}
+	for iteratorNamespaceRules.NotDone() {
+		results = append(results, *(iteratorNamespaceRules.Value()).Name)
+		err = iteratorNamespaceRules.Next()
+		if err != nil {
+			return nil, err
+		}
+	}
+	return results, nil
+}
+
+// ListNamespaceAuthRules - authenticate namespace client and enumerates all values to get list of authorization rules for the given namespace name,
+// automatically crossing page boundaries as required. This function would fail the test if there is an error.
+func ListNamespaceAuthRules(t *testing.T, subscriptionID string, namespace string, resourceGroup string) []string {
+	results, err := ListNamespaceAuthRulesE(subscriptionID, namespace, resourceGroup)
+	require.NoError(t, err)
+
+	return results
+}
+
+// ListNamespaceTopicsE - authenticate topic client and enumerates all values, automatically crossing page boundaries as required.
+func ListNamespaceTopicsE(subscriptionID string, namespace string, resourceGroup string) ([]servicebus.SBTopic, error) {
+	tClient, err := serviceBusTopicClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	iteratorTopics, err := tClient.ListByNamespaceComplete(context.Background(), resourceGroup, namespace, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	results := make([]servicebus.SBTopic, 0)
+
+	for iteratorTopics.NotDone() {
+		results = append(results, iteratorTopics.Value())
+		err = iteratorTopics.Next()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return results, nil
+}
+
+// ListNamespaceTopics - authenticate topic client and enumerates all values, automatically crossing page boundaries as required. This function would fail the test if there is an error.
+func ListNamespaceTopics(t *testing.T, subscriptionID string, namespace string, resourceGroup string) []servicebus.SBTopic {
+	results, err := ListNamespaceTopicsE(subscriptionID, namespace, resourceGroup)
+	require.NoError(t, err)
+
+	return results
+}
+
+// ListTopicSubscriptionsE - authenticate subscriptions client and enumerates all values, automatically crossing page boundaries as required.
+func ListTopicSubscriptionsE(subscriptionID string, namespace string, resourceGroup string, topicName string) ([]servicebus.SBSubscription, error) {
+	sClient, err := serviceBusSubscriptionsClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+	iteratorSubscription, err := sClient.ListByTopicComplete(context.Background(), resourceGroup, namespace, topicName, nil, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	results := make([]servicebus.SBSubscription, 0)
+
+	for iteratorSubscription.NotDone() {
+		results = append(results, iteratorSubscription.Value())
+		err = iteratorSubscription.Next()
```

Comment:
NIT: This should really be `err := iteratorSubscription.Next()`, or even better is:

```
if err := iteratorSubscription.Next(); err != nil {
    return nil, err
}
```

so that it defines a new err var scoped to the forloop/if block. This is a general go pattern to avoid mistakes arising from overriding variables unintentionally.

This applies to the other sections as well.